lnwire/channel_announcement.go

```diff
@@ -184,3 +186,77 @@ func (a *ChannelAnnouncement1) DataToSign() ([]byte, error) {
 
 	return buf.Bytes(), nil
 }
+
+// Validate validates the channel announcement message and checks that node
+// signatures covers the announcement message, and that the bitcoin signatures
+// covers the node keys.
+func (a *ChannelAnnouncement1) Validate(_ func(id *ShortChannelID) (
```

Comment:
so @yyforyongyu  - carrying on from your suggestion re keeping the lnwire pkg purely about encoding & decoding wire messages, do you rate this validation should move to `netann` instead perhaps? 

Then we can maybe just have a single `Validate(ChannelAnnouncement)` function there with a switch on the underlying chan announce msg type?

Reply:
If we move the interface to the `netann` package, then we can have a wrapper struct the accepts either of the wire message types, using the field to implement the new abstracted validation.

Reply:
I think we should also make a type def for the accepted closure, which'll make the signature of the new interface easier to parse.